### ISSUE TITLE
Microsoft has likely acquired GitHub

### ISSUE THREAD

**Issue by yegortimoshenko** [nylira/prism-break#2011]
https://www.theverge.com/2018/6/3/17422752/microsoft-github-acquisition-rumors

In light of this, I would like us to seriously consider self-hosting a Gogs or GitLab instance. We're not eating our own dog food and setting a rather bad example in regards to self-hosting by using GitHub for issues and pull requests.

If it is agreed upon that this is the way to go, I can help to move all of the issues to the new platform, keeping interlinking intact.

This is a duplicate of #767.

cc @nylira @strugee

**Comment by yegortimoshenko** [comment]
They did it: https://blog.github.com/2018-06-04-github-microsoft/

**Comment by 2SI3NX** [comment]
Great!  This https://github.com/nylira/prism-break/issues/2015 is in relation to a warning and maybe a wake up call to all the developers whose software is listed on prism-break

**Comment by strugee** [comment]
@2SI3NX' description from #2015, which I just closed:

> Not sure if you consider this a petty issue, but to many other people and me it seems like this is the ultimate shit storm that will hit all open source code in the very near future.  Do we really want that to happen?
> Since prism-break is a small and precise hub of most privacy related open-source software, i advise we get the attention of at least those software developers to move away from Github and onto an open source alternative that will not be controlled by a giant that already controls half the internet.  
> 
> If we stand together, we can continue our journey of freedom and open source/self-hosted software elsewhere such as Gitlab.
> here is something i stumbled upon on what microsoft is capable of doing just like its done in the past.
> https://jacquesmattheij.com/what-is-wrong-with-microsoft-buying-github
> 
> I respect prism-break for putting together a nice safe list, but it would be for nothing if you do not at least make this apparent to those developers and caution them.

**Comment by strugee** [comment]
So, here are my thoughts on this. I think the hype over this acquisition is _waaaay_ overblown. In recent years (enough of) Microsoft has seemed to really change their culture that I don't think anything really terrible is going to happen to GitHub like people are saying. We the free software community have had problems with GitHub for years and I don't expect them to get any better or worse - just stay the same.

With that being said, GitHub is still obviously proprietary, and that's not so great. So I would be in support of moving to e.g. GitLab, and if people are moving off GitHub because the Microsoft acquisition gave them the push they need, I'm all for it. For this project though I would like us to move not because we worked ourselves up into a frenzy of FUD about Microsoft, but because it is the right thing to do.

**Comment by strugee** [comment]
Let me also clarify before someone bites my head off that I'm not trying to defend anything Microsoft has done in the past; I'm well-aware that they've been terrible. I'm only noting recent trends I see. I also have no illusion that Microsoft has changed purely out of the goodness of their hearts.

**Comment by yegortimoshenko** [comment]
I see this as a sovereignty problem mostly: previously a self-sufficient company was merged into a megacorp, thus giving it power over its users that they are likely to use in self-fulfilling ways. That said, I agree that GitHub was a problem all time along, and this is a good wakeup call.

**Comment by yegortimoshenko** [comment]
@strugee: what do you think about self-hosted GitLab vs GitLab.com?

**Comment by strugee** [comment]
> I see this as a sovereignty problem mostly: previously a self-sufficient company was merged into a megacorp, thus giving it power over its users that they are likely to use in self-fulfilling ways.

Meh. I don't think it's any more likely than before. Besides, I'm not sure GitHub wasn't losing money.  I can't remember at this point tbh. This is not the main point though :P

> @strugee: what do you think about self-hosted GitLab vs GitLab.com?

I would strongly prefer GitLab.com. It's one less account contributors have to manage and it's less ops overhead. I might feel differently if GitLab was federated, but it's not. 

Plus the company behind GitLab IMO has done a really good job managing their free software. They do technically have an open core model but I don't find it all that objectionable since a) it's clear they really tried hard to find a business model where _all_ the code could be libre, b) the Enterprise Edition is as close to libre as they can get it (source code is available and the issue tracker is public) and c) they do a good job keeping the difference between EE and CE small. What do you think?

**Comment by yegortimoshenko** [comment]
cc @Zegnat

**Comment by yegortimoshenko** [comment]
I too think Gitlab.com is more reasonable until federation becomes a thing.

**Comment by yegortimoshenko** [comment]
> Perhaps we can give everyone a week to create GitHub linked accounts before doing the move, so we can save as much history and commit attribution as possible.

I agree. I've just made a test import and there are quite a few accounts that were not recognized.

I'll mention the most active ones that are missing: @nylira, @Zegnat, @JeremyRand, @Atavic, @anthologist, @vyp, @Cathryne, @cyanotex, @quantumpacket, @jinformatique. 

If possible, please create a GitLab account by June 12, and connect it to GitHub profile (this step is crucial). I see that some already have a GitLab account, just that it's not connected to one on GitHub.

**Comment by JonnyTech** [comment]
> If possible, please create a GitLab account by June 12, and connect it to GitHub profile (this step is crucial)

Quick link to do so: log into GitLab then goto https://gitlab.com/users/auth/github

**Comment by ohmynameisrico** [comment]
Not everyone noticed Gitlab infrastructure is behind Microsoft Azure lol
https://twitter.com/gnaggnoyil/status/1003656811172413440

**Comment by quantumpacket** [comment]
They spoke about that on their livestream. They are migrating to Google Cloud...no better, but at least people can self-host a GitLab instance if they choose to do so.

**Comment by ohmynameisrico** [comment]
^ edit: command to test: "traceroute gitlab.net"

**Comment by ohmynameisrico** [comment]
I prefer self-host with .onion(Tor Onion Service), not rely on MS Azure(gitlab).

**Comment by yegortimoshenko** [comment]
@ohmynameisrico We still have to take contributior's convenience into account, which largely was the reason why we were stuck with GitHub all this time along. Unfortunately, asking every contributor to sign up on our instance specifically to contribute to PRISM Break is less than reasonable.

I very much look forward to federation (https://gitlab.com/gitlab-org/gitlab-ee/issues/4517).

**Comment by yegortimoshenko** [comment]
@quantumpacket Maybe we shouldn't rely on proprietary GitLab features :-) Free GitLab is roughly at feature parity with GitLab CE, meaning easy migration if necessary, and that covers everything GitHub does, and that was sufficient all along.

**Comment by quantumpacket** [comment]
I agree, but we don't need to use features that could introduce migration issues, but take advantage of other things like audit logs, verified commiter,  etc, which comes with the higher tiers.

**Comment by ohmynameisrico** [comment]
http://gitweb2zl5eh7tp3.onion/ ( https://www.reddit.com/r/onions/comments/3ykctz/free_git_repository_hosting_accessible_only_over/ )
and
http://haklab4ulibiyeix.onion/users/sign_in

**Comment by yegortimoshenko** [comment]
> I have signed up for GitLab, grabbing @Zegnat, and linked it to my GitHub. (Although I may or may not have rolled my eyes a little at jumping from one hosted service onto another. :wink:)

We'll have easier time migrating from GitLab.com to self-hosted GitLab because import will be lossless (any repo can be exported via Settings -> General -> "Export project" option). Also, we as users will be much more in control because of source code availability for the version that we're going to migrate to and the ability to self-host at our convenience if things go wrong. Of course I agree this is not an ideal solution, but it seems to strike a balance between practicality and goals of this project than GitHub. By the way, they don't use analytics on their site (this place uses Google Analytics), all their JS is under a free license (even in EE edition),

**Comment by yegortimoshenko** [comment]
I'm going to start migration on Saturday, unless anyone minds.

**Comment by beerisgood** [comment]
Gitlab? Kidding right? Google will buy Gitlab in near future. (https://github.com/privacytoolsIO/privacytools.io/issues/478#issuecomment-394891436)
And then? 
No. If you want leave Github just because nothing change, but the owner is Microsoft then use selfhosted.

Also you guys here know what Gitlab is currently behind Microsoft Azure, right?: https://twitter.com/gnaggnoyil/status/1003656811172413440

**Comment by yegortimoshenko** [comment]
It is a better compromise than GitHub, because they are interested in federation very much unlike GitHub, and we can always fall back to self-hosting a GitLab instance.

**Comment by yegortimoshenko** [comment]
Moved to: http://gitlab.com/prism-break/prism-break/

**Comment by Hillside502** [comment]
Think you mean:-
https://gitlab.com/prism-break/prism-break/

**Comment by uBlock-user** [comment]
@beerisgood https://about.gitlab.com/2018/06/25/moving-to-gcp/

**Comment by beerisgood** [comment]
Yes and we all say that but No one cares. Let's see how much come back to GitHub

**Comment by kradan** [comment]
I tried to register at gitlab.com but failed because the captcha was not shown using TBB 8 with extensions. It was necessary to disable privacybadger and allow gstatic.com for requestblock. Then finally google told me:
> Please upgrade to a [supported](https://support.google.com/recaptcha/?hl=en#6223828) browser to get a reCAPTCHA challenge.
> Alternatively if you think you are getting this page in error, please check your internet connection and reload.
I never had such issues at github.

[Gitlab](https://en.wikipedia.org/wiki/GitLab#History) adopted [open-core](https://en.wikipedia.org/wiki/Open_core) in 2014. As pointed out above, there's no fundamental difference to github and we can expect it to be sold pretty soon as well.

[Jun 25, 2018 - Andrew Newdigate We’re moving from Azure to Google Cloud Platform. GitLab.com is migrating to Google Cloud Platform – here’s what this means for you now and in the future. 
](https://about.gitlab.com/2018/06/25/moving-to-gcp/)

however it seems to be hosted by microsoft just as github
```
traceroute to gitlab.com (52.167.219.168), 30 hops max, 60 byte packets                                         
 ...
 8  microsoft.bcix.de (193.178.185.84)  73.603 ms  78.654 ms  78.876 ms                                         
 9  ae22-0.amb-96cbe-1b.ntwk.msn.net (104.44.224.99)  89.858 ms  90.915 ms  90.200 ms                           
10  be-62-0.ibr01.amb.ntwk.msn.net (104.44.9.134)  175.785 ms  182.743 ms  181.174 ms                           
11  be-5-0.ibr01.lts.ntwk.msn.net (104.44.4.233)  181.389 ms  182.430 ms  181.960 ms                            
12  be-1-0.ibr02.lts.ntwk.msn.net (104.44.4.220)  192.917 ms be-2-0.ibr01.lon30.ntwk.msn.net (104.44.5.38)  198.148 ms  171.342 ms
13  be-2-0.ibr02.lon30.ntwk.msn.net (104.44.5.40)  176.920 ms  177.407 ms be-11-0.ibr01.nyc30.ntwk.msn.net (104.44.5.104)  177.640 ms
14  be-1-0.ibr02.nyc30.ntwk.msn.net (104.44.7.206)  177.005 ms  178.231 ms  177.946 ms                          
15  be-6-0.ibr02.was05.ntwk.msn.net (104.44.4.29)  178.643 ms  181.699 ms  182.025 ms                           
16  be-9-0.ibr01.cnr02.bn6.ntwk.msn.net (104.44.4.26)  192.928 ms  193.084 ms  193.486 ms                       
17  ae120.0.icr01.bn6.ntwk.msn.net (104.44.10.51)  187.143 ms ae121-0.icr03.bn6.ntwk.msn.net (104.44.10.128)  192.115 ms ae123-0.icr04.bn6.ntwk.msn.net (104.44.10.130)  193.808 ms
```
[msn.net = msn.com](https://en.wikipedia.org/w/index.php?title=Msn.net) or what am I missing?

**Comment by kradan** [comment]
I did not know of people having captcha issues with github, just because I did not experience them myself. To assume there are none was ignorant by me. From your answer I read, that you are very happy with the change to gitlab. So you do not want it criticized, right?

The original reason for prism-break from my understanding was the "news" that big companies were working together or made a valuable target for agencies just because they are centralizing information ([SPOF](https://en.wikipedia.org/wiki/Single_point_of_failure)) which resulted in the motivation to provide "[A list of alternatives to software and systems that are vulnerable to eavesdropping](https://en.wikipedia.org/wiki/PRISM_(surveillance_program)#External_links)". This was in 2013, 6 years after the program had begun, now 5 years later we have this list but are relying on hosting services that aren't better.

Hence I disagree, that it is irrelevant where our common repository is hosted.

Yes, gitlab may be a progress compared to github and my point is not "switch back", rather:  

My points are:
1) don't forget people not at gitlab
2) Git is designed for decentralization and we should try to make benefit of it.
3) a service [openly endorsing cloud services](https://about.gitlab.com/2018/06/25/moving-to-gcp/) and requiring google captchas to register is the opposite of breaking free from [PRISM](https://en.wikipedia.org/wiki/PRISM_(surveillance_program))
4) gitlab is no safe harbour and we should continue looking for alternatives (as pointed out in the original #767) like an onion based CDN (which to my knowledge does not exist yet)

Please make a full stop before answering if you did not realize that yourself before. When you accuse me of spreading [FUD](https://en.wikipedia.org/wiki/FUD) as in fear, uncertainty and doubt I feel deeply misunderstood as I pointed out a speculation, shared my experience and asked a question (which you did not answer).

@alerque, please don't feel attacked! The proposal you made unfortunately does not work for me, because of the services you listed I only use github (obviously). Now I am left with no choice than connecting my github account with a new gitlab account. I am not sure if I want to do it.

I am advocating for us to set up many mirrors of this repository and to get used to do this for all the code we value. Is this a silly idea?

**Comment by yegortimoshenko** [comment]
> 1. don't forget people not at gitlab

That can be said about anything remotely social that we list (e.g. Diaspora, Mastodon) in comparison to what most people are using (e.g. Facebook, Twitter).

> 2. Git is designed for decentralization and we should try to make benefit of it.

Git doesn't really matter much here, if it did, you wouldn't state point 1. Most of PRISM Break's work happens in issue tracker. Previously we used a proprietary vendor-locked issue tracker, now we get flexibility to switch providers. It also doesn't track (GitHub uses Google Analytics). Seems like a net benefit overall.

> 3. a service openly endorsing cloud services and requiring google captchas to register is the opposite of breaking free from PRISM

While I agree CAPTCHAs are very problematic, if you're comfortable with GitHub (you seem to be), you can just login into GitLab via GitHub. We are not stating GitLab.com is perfect, but it's hard to argue it's not an upgrade over using this place.

> Now I am left with no choice than connecting my github account with a new gitlab account.

I can register a GitLab account with a nickname of your choice and a temporary password and send it to you.

> I am advocating for us to set up many mirrors of this repository and to get used to do this for all the code we value. Is this a silly idea?

Feel free to mirror. Upstream is at https://gitlab.com/prism-break/prism-break.

**Comment by Hillside502** [comment]
> GitHub uses Google Analytics

and their own `collector.githubapp.com`
